src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/azure/vm/AzureVMsProvider.java

```diff
@@ -0,0 +1,295 @@
+import org.ow2.proactive.connector.iaas.cloud.provider.azure.AzureProviderUtils;
+import org.ow2.proactive.connector.iaas.cloud.provider.azure.AzureServiceCache;
+import org.ow2.proactive.connector.iaas.model.*;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.Lists;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.compute.*;
+import com.microsoft.azure.management.network.*;
+import com.microsoft.azure.management.network.Network;
+import com.microsoft.azure.management.network.model.HasPrivateIpAddress;
+import com.microsoft.azure.management.network.model.HasPublicIpAddress;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.azure.management.resources.fluentcore.model.Creatable;
+import com.microsoft.azure.management.resources.fluentcore.utils.SdkContext;
+
+import lombok.Getter;
+
+
+/**
+ * Provides Microsoft Azure clouds' management using the official java SDK.
+ *
+ * This class has been tested by ActiveEon to be thread safe (using Azure SDK release version 1.0.0-beta5).
+ * However this need to be carefully double-checked after every SDK upgrades, as mentioned by Microsoft:
+ * ------------------------------------------------------------------------------------------------------------
+ * We do not make any thread-safety guarantees about our libraries. We also do not test them for thread-safety.
+ * Methods that are currently thread-safe may be thread-unsafe in future versions.
+ * ------------------------------------------------------------------------------------------------------------
+ *
+ * @author ActiveEon Team
+ * @since 01/03/17
+ */
+@Component
+public class AzureVMsProvider extends AzureProvider {
+
+    @Getter
+    protected final String type = "azure";
+
+    private final Logger logger = Logger.getLogger(AzureVMsProvider.class);
+
+    @Override
+    public Set<Instance> createInstance(Infrastructure infrastructure, Instance instance) {
+
+        Azure azureService = azureServiceCache.getService(infrastructure);
+        String instanceTag = Optional.ofNullable(instance.getTag())
+                                     .orElseThrow(() -> new RuntimeException("ERROR missing instance tag/name from instance: '" +
+                                                                             instance + "'"));
+
+        // Check for Image by name first and then by id
+        String imageNameOrId = Optional.ofNullable(instance.getImage())
+                                       .orElseThrow(() -> new RuntimeException("ERROR missing Image name/id from instance: '" +
+                                                                               instance + "'"));
+
+        // TODO: no custom image allowed currently
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1135)